src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -95,6 +96,12 @@ public GetSecretValueResponse handleRequest(GetSecretValueRequest request) {
             }
         }
 
+        @Override
+        public CompletableFuture<GetSecretValueResponse> handleRequestAsync(GetSecretValueRequest request) {
+            // TODO: Limit the no of simultaneous requests using a separate threadpool with configurable size
+            return CompletableFuture.supplyAsync(() -> handleRequest(request));
```

Comment:
as long as it's added before release, we'll need a executor otherwise it'll use forkjoinpool